### ISSUE TITLE
Add Shield Stats

### ISSUE THREAD

**Issue by kylehickinson** [brave/brave-ios#161]
Currently no shield stats are persisted or accumulating.

Known affected Files: `BraveShieldState.swift`, `ShieldsViewController.swift`, `BraveShieldStatsView.swift`, `BrowserViewController.swift`

1. Move the stat counts storage to `Preferences` (which may mean bringing the base implementation of `Preferences` to `BraveShared`, or pulling counts out entirely from `BraveShieldState` to `Client`)

2. Add the **local** counts back to the shields panel (`ShieldsViewController.init(url: URL, shieldBlockStats: ShieldBlockedStats?)`). These need to be updated as shields are updated. This may be blocked on other shield related items and require a new ticket down the road

3. Add the counts back to the home panel for new tabs in `BraveShieldStatsView`

4. Also looks like theres something to do in Domain.swift: setBraveShield(forDomain:state:context:) and loadShieldsIntoMemory()

**Comment by kylehickinson** [comment]
Also looks like theres something to do in `Domain.swift`: `setBraveShield(forDomain:state:context:)` and `loadShieldsIntoMemory()`
